workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/DialogSession.kt

```diff
@@ -0,0 +1,160 @@
+package com.squareup.workflow1.ui.container
+
+import android.os.Bundle
+import android.os.Parcel
+import android.os.Parcelable
+import android.os.Parcelable.Creator
+import android.view.KeyEvent
+import android.view.MotionEvent
+import android.view.Window
+import androidx.core.view.doOnAttach
+import androidx.core.view.doOnDetach
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import com.squareup.workflow1.ui.Compatible
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.androidx.WorkflowLifecycleOwner
+import com.squareup.workflow1.ui.androidx.WorkflowSavedStateRegistryAggregator
+
+/**
+ * Used by [LayeredDialogSessions] to manage lifecycle and view persistence concerns for an
+ * [OverlayDialogHolder], as well as enforcing modal behavior.
+ */
+@WorkflowUiExperimentalApi
+internal class DialogSession(
```

Comment:
What I found I wanted on these Session classes is a definition of the lifecycle of a session. I know its internal and an implementation detail - but I still would love to know :).